cake/build-settings.cake

```diff
@@ -1,23 +1,354 @@
+        UnitTestRunner unitTestRunner = null, // If not set, NUnitLite is used
+        string unitTestArguments = null
+        )
+	{
+        // Required arguments
         if (context == null)
-            throw new System.ArgumentNullException(nameof(context));
+            throw new ArgumentNullException(nameof(context));
+		if (title == null)
+			throw new ArgumentNullException(nameof(title));
+		if (githubRepository == null)
+			throw new ArgumentNullException(nameof(githubRepository));
+
+        Context = context;
+        Title = title;
+        GitHubRepository = githubRepository;
+
+		// NOTE: Order of initialization can be sensitive. Obviously,
+		// we have to set any properties in this method before we
+		// make use of them. Less obviously, some of the classes we
+		// construct here have dependencies on certain properties
+		// being set before the constructor is called. I have
+		// tried to annotate such dependencies below.
+
+        _buildSystem = context.BuildSystem();
+	
+		// If not specified, uses TITLE.sln if it exists or uses solution
+		// found in the root directory provided there is only one. 
+		SolutionFile = solutionFile ?? DeduceSolutionFile();
+
+		ValidConfigurations = validConfigurations ?? DEFAULT_VALID_CONFIGS;
+
+		UnitTests = unitTests;
+		// NUnitLiteRunner depends indirectly on ValidConfigurations
+		UnitTestRunner = unitTestRunner ?? new NUnitLiteRunner();
+		UnitTestArguments = unitTestArguments;
+
+		BuildVersion = new BuildVersion(context);
+
+        GitHubOwner = githubOwner;
+
+		// File Header Checks
+		SuppressHeaderCheck = suppressHeaderCheck && !CommandLineOptions.NoBuild;
+		StandardHeader = standardHeader ?? DEFAULT_STANDARD_HEADER;
+		ExemptFiles = exemptFiles ?? new string[0];
+        
+		//if (defaultTarget != null)
+		//	BuildTasks.DefaultTask.IsDependentOn(defaultTarget);
+
+		// Skip remaining initialization if help was requested
+		if (CommandLineOptions.Usage)
+			return;
+			
+		ValidateSettings();
 
-        Target = context.TargetTask.Name;
-        Configuration = context.Argument("configuration", "Release");
+		context.Information($"{Title} {Configuration} version {PackageVersion}");
+
+		// Output like this should go after the run title display
+		if (solutionFile == null && SolutionFile != null)
+			Context.Warning($"  SolutionFile: '{SolutionFile}'");
+		Context.Information($"  PackageTestLevel: {PackageTestLevel}");
+
+		// Keep this last
+		if (IsRunningOnAppVeyor)
+		{
+			var buildNumber = _buildSystem.AppVeyor.Environment.Build.Number;
+			_buildSystem.AppVeyor.UpdateBuildVersion($"{PackageVersion}-{buildNumber}");
+		}
     }
 
-    public string Target { get; }
-    public string Configuration { get; }
+	// Try to figure out solution file when not provided
+	private static string DeduceSolutionFile()			
+	{
+		string solutionFile = null;
+
+		if (System.IO.File.Exists(Title + ".sln"))
+			solutionFile = Title + ".sln";
```

Comment:
I would just return here directly instead of do the single return at the bottom.  You're already using multiple returns in CalcPackageTestLevel so there's precedent for it, and this way makes it look like there might be other "processing" of `solutionFile` before returing.